src/app/3d/qgs3dmapcanvaswidget.cpp

```diff
@@ -305,6 +308,19 @@ Qgs3DMapCanvasWidget::Qgs3DMapCanvasWidget( const QString &name, bool isDocked )
 
   onTotalPendingJobsCountChanged();
 
+  QAction *toggleDebugPanel = toolBar->addAction(
```

Comment:
I don't think we should make this visible by default. It's not something the average user ever needs, and adding it to the toolbar just contributes to UI clutter. Instead of a toolbar action, how about some hidden key shortcut to toggle the pane?

Reply:
I agree, but I'd rather have a `Show debug panel` checkbox in advanced settings page instead of completely hiding it!

Reply:
I like the @uclaros approach of having a checkbox + I can add a keyboard shortcut

Reply:
I moved the setting into advanced sub-category of options and added a Ctrl+Shift+d shortcut for the panel

Reply:
@Withalion 
> Ctrl+Shift+d 

That's already used for opening Data Source Manager. How about F12? That's used for the dev tools panel in the main window, so I think it'd be nice to use that here too. (I realise that if the 3d view is docked then F12 would only work if the 3d view is focused, but that same behavior would also be present with Ctrl+Shift+D due to the conflict with the Data Source Manager shortcut)

Reply:
Okay that makes sense. I guess having an overlap in shortcuts is unavoidable, but it's definitely better UX on F12.

Reply:
@nyalldawson after testing the change, F12 on docked windows doesn't work at all. For the previous combination I at least got a window about shortcut ambiguity, but on F12 I don't get anything. It just doesn't work. In stand alone window it works.